### PR TITLE
Add nodal type definitions.

### PR DESCRIPTION
**Pull request by charrondev**
Please fill in this template.
- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If adding a new definition:
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Run `tsc` without errors.
- [x] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header.

If changing an existing definition:
- [ ] Provide a URL to  documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by vvakame** [comment]
why split nodal/index.d.ts and nodal/fxn.d.ts?

and could you fix travis ci failed?

File: nodal/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1010 @@
+// Type definitions for nodal 0.12.8
+// Project: https://github.com/keithwhor/nodal
+// Definitions by: Adam Charron <https://github.com/charrondev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../node/index.d.ts"/>
```
Comment (Charrondev) [inline nodal/index.d.ts]: Ah thank you! I'll fix it.

**Comment by Charrondev** [comment]
nodal and fxn are split up because fxn is actually a separate npm package that nodal depends on, but I only typed the relevant parts that make it into the public API. What I don't understand is why the Travis CI build is failing. All the failures are from typing that I made no changes to (zepto, bardjs, angular). What could be causing that?

**Comment by vvakame** [comment]
sorry, Travis CI failed that is a our problem.
types-2.0/HEAD is back to green maybe...

**Comment by vvakame** [comment]
fxn.d.ts is https://www.npmjs.com/package/fxn ?
please separate directly by 1 npm package.

**Comment by minestarks** [comment]
@Charrondev Thank you for your contribution. Do you plan to address the requested changes?

**Comment by Charrondev** [comment]
Yes I will split them up. It will probably be submitted later today.

**Comment by Charrondev** [comment]
This should be fine now. Tests are passing, and the extra dep has been split into its own folder and named appropriately.

**Pull request merged**